### ISSUE TITLE
using render? in component causes issues

### ISSUE THREAD

**Issue by Atrox** [unabridged/motion#14]
Hey,

when using `render?` and it evaluates to `false` in a component the following error occurs on render ```undefined method `[]=' for nil:NilClass```. It works fine in a normal view component.

Example:
```ruby
class TestComponent < ViewComponent::Base
  include Motion::Component

  def initialize(test:)
    @test = test
  end

  def render?
    false
  end
end
```

```erb
<%= render(TestComponent.new(test: 'test')) %>
```

```
ActionView::Template::Error (undefined method `[]=' for **nil:NilClass):**
```

While I have the chance, I just wanna say thank you for building this. I'm just playing with it for now but it feels really good :).

**Comment by alecdotninja** [comment]
That's a good point. Thank you for pointing this out, @Atrox !

You are right. This feature is currently broken :see_no_evil:

**Comment by alecdotninja** [comment]
Hmm, this is an unexpectedly nuanced issue. The exception given on master explains what is going on a bit better:

```
[CalculatorComponent:25200] An error occurred while rendering the component:
	Motion::MultipleRootsError: The template for CalculatorComponent can only have one root element.

	Fix: Wrap all elements in a single element, such as <div> or <section>.

		/home/alec.larsen/Development/motion/lib/motion/markup_transformer.rb:39:in `transform_root'
		/home/alec.larsen/Development/motion/lib/motion/markup_transformer.rb:26:in `add_state_to_html'
		/home/alec.larsen/Development/motion/lib/motion/component/rendering.rb:45:in `render_in'
		/home/alec.larsen/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/view_component-2.8.0/lib/view_component/render_to_string_monkey_patch.rb:7:in `render_to_string'
		/home/alec.larsen/.asdf/installs/ruby/2.7.1/lib/ruby/gems/2.7.0/gems/actionpack-6.0.3.2/lib/action_controller/renderer.rb:96:in `render'
```

The problem is that Motion currently depends on having a single root element, and when you render nothing you have no root element. The fixed proposed in #15 is good for the initial render, but I am torn about the case where the component is already connected.

I think the easiest thing to do technically is to remove the component from the DOM (and disconnect in the process). This is also consistent with what happens on the initial render, but the semantics are a little weird IMHO.
